app/uk/gov/hmrc/incometaxsessiondata/services/SessionService.scala

```diff
@@ -17,27 +17,28 @@
 package uk.gov.hmrc.incometaxsessiondata.services
 
 import uk.gov.hmrc.incometaxsessiondata.domain.models.Session
+import uk.gov.hmrc.incometaxsessiondata.repositories.SessionDataRepository
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class SessionService @Inject(){
+class SessionService @Inject()(
+                                repository: SessionDataRepository
+                              ) {
```

Comment:
worth to expose execution on the class level, ie: and remove one we have on the method(s)
```
class SessionService @Inject()(repository: SessionDataRepository)
 (implicit ec: ExecutionContext){
```